controllers/placementapi_controller.go

```diff
@@ -207,6 +333,8 @@ func (r *PlacementAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.ServiceAccount{}).
 		Owns(&rbacv1.Role{}).
 		Owns(&rbacv1.RoleBinding{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.GetSecretMapperFor(&placementv1.PlacementAPIList{}, context.TODO()))).
```

Comment:
This is in conflict with https://github.com/openstack-k8s-operators/placement-operator/pull/92/files#diff-c0f1c404ea10bc348fba06aab6f455b95b7513a337e09b325ebea615fc0ee587R222 

As far as I understand the pattern for watching for Secret objects in https://github.com/openstack-k8s-operators/placement-operator/pull/92 is more efficient. 

So we need to figure out a way forward that incorporate both changes as this PR has ensureSecret which is good and #92 has an advanced Secret watching solution.

Reply:
// later
It seem the two watch can coexists based on the proposal in https://github.com/openstack-k8s-operators/nova-operator/pull/646/files#diff-acc4b831bcb31906497644cbef12aa82d90a4a6690078596ca10437735c05e11R850-R856

Reply:
Let land this independently from TLS. And after both landed we can try to combine the two watch